src/TorchSharp/Utils/ShuffleGenerator.cs

```diff
@@ -0,0 +1,89 @@
+using System;
+
+namespace TorchSharp.Utils
+{
+    public class ShuffleGenerator
+    {
+        int maxrange;
+        int prime;
+        int index;
+        int offset;
+        int runningvalue;
+
+        public ShuffleGenerator(int size)
```

Comment:
Please add a reference to where the algorithm comes from, even if it's just a Pytorch source file. A paper describing it would be best, but some attribution is important.